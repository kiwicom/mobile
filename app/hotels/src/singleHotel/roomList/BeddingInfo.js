// @flow

import * as React from 'react';
import { View } from 'react-native';
import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
import { StyleSheet, Text, TextIcon } from '@kiwicom/mobile-shared';
import { Translation } from '@kiwicom/mobile-localization';
import idx from 'idx';
import { defaultTokens } from '@kiwicom/mobile-orbit';

import type { BeddingInfo_room } from './__generated__/BeddingInfo_room.graphql';
import formatBeddingInfo from './formatBeddingInfo';

type ContainerProps = {|
  room: any,
|};

type Props = {
  ...ContainerProps,
  room: ?BeddingInfo_room,
};

export class BeddingInfo extends React.Component<Props> {
  render = () => {
    const { room } = this.props;
    const info = formatBeddingInfo(room);
    const maxPersons = idx(this.props.room, _ => _.maxPersons);

    return (
      <View>
        <View style={styles.row}>
          <TextIcon code="&#xe085;" style={styles.icon} />
          <Text style={styles.text}>
            <Translation passThrough={` ${info}`} />
          </Text>
        </View>
        <View style={styles.row}>
          <TextIcon code="(" style={styles.icon} />
          <Text style={styles.text}>
            <Translation passThrough={' '} />
            <Translation
              id="single_hotel.bedding_info.guests"
              values={{ numberOfGuests: maxPersons }}
            />
          </Text>
        </View>
      </View>
    );
  };
}

export default (createFragmentContainer(
  BeddingInfo,
  graphql`
    fragment BeddingInfo_room on HotelRoom {
      maxPersons
      bedding {
        type
        amount
      }
    }
  `,
): React.ComponentType<ContainerProps>);

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    marginBottom: 5,
  },
  text: {
    fontSize: 12,
    lineHeight: 15,
    color: defaultTokens.colorTextSecondary,
  },
  icon: {
    color: defaultTokens.colorTextSecondary,
    fontSize: 14,
    marginEnd: 2,
    alignSelf: 'center',
  },
});
