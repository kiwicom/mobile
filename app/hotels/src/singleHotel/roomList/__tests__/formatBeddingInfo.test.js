// @flow

import { BeddingInfo } from '../BeddingInfo';

it('formats bedding information', () => {
  // $FlowRelayIssue: https://github.com/facebook/relay/issues/2394
  const Component = new BeddingInfo({
    room: {
      type: 'Single Room',
      maxPersons: 1,
      bedding: [
        {
          type: 'Single Bed(s)',
          amount: 1,
        },
      ],
    },
  });
  expect(Component.formatBeddingInfo()).toMatchSnapshot();
});

it('pluralizes persons', () => {
  // $FlowRelayIssue: https://github.com/facebook/relay/issues/2394
  const Component = new BeddingInfo({
    room: {
      type: 'Double Room',
      maxPersons: 2,
      bedding: [
        {
          type: 'Single Bed(s)',
          amount: 2,
        },
      ],
    },
  });
  expect(Component.formatBeddingInfo()).toMatchSnapshot();
});

it('provides all bedding options', () => {
  // $FlowRelayIssue: https://github.com/facebook/relay/issues/2394
  const Component = new BeddingInfo({
    room: {
      type: 'Double or Twin Room',
      maxPersons: 2,
      bedding: [
        {
          type: 'Single Bed(s)',
          amount: 2,
        },
        {
          type: 'Twin Bed(s)',
          amount: 1,
        },
      ],
    },
  });
  expect(Component.formatBeddingInfo()).toMatchSnapshot();
});

it("doesn't crash on empty input", () => {
  expect(new BeddingInfo({ room: null }).formatBeddingInfo()).toMatchSnapshot();
  expect(
    // $FlowRelayIssue: https://github.com/facebook/relay/issues/2394
    new BeddingInfo({
      room: {
        type: 'Some Room',
        maxPersons: 2,
        bedding: null,
      },
    }).formatBeddingInfo(),
  ).toMatchSnapshot();
});
