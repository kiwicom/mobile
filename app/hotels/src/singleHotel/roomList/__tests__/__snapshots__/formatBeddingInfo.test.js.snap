// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't crash on empty input 1`] = `""`;

exports[`formats bedding information 1`] = `"1 Single Bed(s)"`;

exports[`pluralizes persons 1`] = `"2 Single Bed(s)"`;

exports[`provides all bedding options 1`] = `"2 Single Bed(s) or 1 Twin Bed(s)"`;
